src/core/ext/transport/chttp2/transport/hpack_parser.cc

```diff
@@ -1602,6 +1602,7 @@ void grpc_chttp2_hpack_parser_init(grpc_chttp2_hpack_parser* p) {
   p->on_header = nullptr;
   p->on_header_user_data = nullptr;
   p->state = parse_begin;
+  p->log_enabled = GRPC_TRACE_FLAG_ENABLED(grpc_http_trace);
```

Comment:
@yashykt do you have any objections to use `GRPC_DEBUG_TRACE_FLAG`?

Reply:
@roth same question

Reply:
I do unfortunately :) http traces are useful in knowing what was received by gRPC. I'm not sure if users will always be able to switch to debug build for logs.

Reply:
just for hpack parser, i would be fine with debug logs

Reply:
Ah ok great.  So, `hpack_parser` alone would be huge savings. :-)

Reply:
So the point of this PR is a compromise - it means that we still do have non-debug logs, but it does mean that we only get logs on new connections after logging is enabled (unless it was enabled at init time, in which case all connections get it).

Now, I will say that both on_initial_header and on_trailing_header also appear to perform a log.

@yashykt : WDYT?

Reply:
@arjunroy do you mean using a debug-only trace on `on_initial_header` and `on_trailing_header` too?

Reply:
@soheilhy I actually did not plan to change those in this PR - since it seems people want at least some run-time enable-able logging.

Reply:
So, since @yashykt is fine with debug-tracing for just hpack_parser - we can leave on_initial/trailing_header logging as is, and make it debug only for hpack_parser - If both of you (@yashykt , @soheilhy ) are fine with it, I'll change it to do that.

Reply:
+1 let's do that.  Then we can follow up separately for `on_initial_header` `on_trailing_header`